app/protected/solve/[problemId]/[methodId]/page.tsx

```diff
@@ -82,21 +84,44 @@ export default function SolvingPage() {
         if (isChatOpen) setChatClosed(false);
     }, [isChatOpen]);
 
-    // Listen for the chat-toggle event
-    useEffect(() => {
-        const handler = () => setIsChatOpen((v) => !v);
-        window.addEventListener('chat-toggle', handler as EventListener);
-        return () =>
-            window.removeEventListener('chat-toggle', handler as EventListener);
-    }, []);
-
     const handleNextStep = () => {
         if (currentStep < totalSteps - 1) {
-            setCurrentStep((prev) => prev + 1);
+            const nextStep = currentStep + 1;
+            setCurrentStep(nextStep);
+            void tracked.logEvent({
+                actionName: 'next_step',
+                payload: { total_steps: totalSteps },
+                stepId: nextStep,
+            });
         }
     };
 
+    const handleGoToAnswer = () => {
+        setIsAnswerPopupOpen(true);
+        void tracked.logEvent({
+            actionName: 'go_to_answer',
+            stepId: currentStep,
+            payload: { total_steps: totalSteps },
+        });
+    };
+
+    // Open chat and log
+    const handleOpenChat = () => {
+        setIsChatOpen(true);
+        void tracked.logEvent({ actionName: 'chat_open', payload: {} });
+    };
+
+    // Close chat and log
+    const handleCloseChat = () => {
+        setIsChatOpen(false);
+        void tracked.logEvent({ actionName: 'chat_close', payload: {} });
+    };
```

Comment:
See the comment below.
```suggestion
    const handleCloseChat = () => {
        setShowToggle(false);
        setIsChatOpen(false);
        void tracked.logEvent({ actionName: 'chat_close', payload: {} });
    };
```